internal/handlers/images.go

```diff
@@ -59,12 +63,30 @@ func NewImageHandler(is imagestore.ImageStore, assistedServiceScheme, assistedSe
 		Recorder: metrics.NewRecorder(metricsConfig),
 	})
 
+	client := http.DefaultClient
+	if caCertFile != "" {
+		caCert, err := ioutil.ReadFile(caCertFile)
+		if err != nil {
+			log.Fatalf("Error opening cert file %s, %s", caCertFile, err)
+		}
+		caCertPool := x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(caCert)
```

Comment:
It looks like [AppendCertsFromPEM](https://pkg.go.dev/crypto/x509#CertPool.AppendCertsFromPEM) returns a boolean to indicate whether it succeeded or not.

What do you think about consuming the return value and print a warning/error in case of failure?

Reply:
My thinking is that if we can't add the cert, we should fail since there should only ever be one that we are adding **and** we obviously need it.

Reply:
> My thinking is that if we can't add the cert, we should fail since there should only ever be one that we are adding and we obviously need it.

Failing in this case makes sense. Are you going to do the change in this PR?

Reply:
I'm glad you asked that because I thought I had done it already.